apiserver/facades/client/charms/charmhubrepo.go

```diff
@@ -276,7 +274,11 @@ func (c *chRepo) selectNextBase(bases []transport.Base, origin corecharm.Origin)
 		return corecharm.Platform{}, errors.NotFoundf("base")
 	}
 
-	series, err := coreseries.VersionSeries(channelTrack(base.Channel))
+	track, err := channelTrack(base.Channel)
+	if err != nil {
+		return corecharm.Platform{}, errors.Trace(err)
+	}
+	series, err := coreseries.VersionSeries(track)
```

Comment:
I still think we need to check how CentOS and Windows charms will appear as bases. Seems to me we are unlikely to get `win10` or `centos7` as a track. Something like `name: "windows"` and `track: "10"` would be the correct representation.

Reply:
There are currently no way to tell, as we don't have any pre-existing charms to test.